resources/[system]/scoreboard/ui/utils/useNuiEvent.ts

```diff
@@ -0,0 +1,47 @@
+import { MutableRefObject, useEffect, useRef } from 'react';
+import { eventNameFactory } from './tools/nuiUtils';
+
+interface IOptions {
+    capture?: boolean;
+    passive?: boolean;
+    once?: boolean;
+}
+/**
+ * A hook that manage events listeners for receiving data from the NUI
+ * @param app The app name in which this hoook is used
+ * @param method The specific `method` field that should be listened for.
+ * @param handler The callback function that will handle data relayed by this hook
+ * @param options Any options to pass to the addEventListener
+ **/
+const defaultOptions = {};
+
+export const useNuiEvent = <S = Record<string, unknown>>(
```

Comment:
Cute. Where'd this get copied from?

Reply:
An resource I am currently working on.

Link: https://github.com/project-error/new-phone-who-dis/blob/master/phone/src/os/nui-events/hooks/useNuiEvent.ts